packages/ripple-binary-codec/src/enums/index.ts

```diff
@@ -1,164 +1,34 @@
 import * as enums from './definitions.json'
-import { SerializedType } from '../types/serialized-type'
-import { Buffer } from 'buffer/'
-import { BytesList } from '../binary'
-
-/*
- * @brief: All valid transaction types
- */
-export const TRANSACTION_TYPES = Object.entries(enums.TRANSACTION_TYPES)
-  .filter(([_key, value]) => value >= 0)
-  .map(([key, _value]) => key)
-
-const TYPE_WIDTH = 2
-const LEDGER_ENTRY_WIDTH = 2
-const TRANSACTION_TYPE_WIDTH = 2
-const TRANSACTION_RESULT_WIDTH = 1
-
-/*
- * @brief: Serialize a field based on type_code and Field.nth
- */
-function fieldHeader(type: number, nth: number): Buffer {
-  const header: Array<number> = []
-  if (type < 16) {
-    if (nth < 16) {
-      header.push((type << 4) | nth)
-    } else {
-      header.push(type << 4, nth)
-    }
-  } else if (nth < 16) {
-    header.push(nth, type)
-  } else {
-    header.push(0, type, nth)
-  }
-  return Buffer.from(header)
-}
-
-/*
- * @brief: Bytes, name, and ordinal representing one type, ledger_type, transaction type, or result
- */
-export class Bytes {
-  readonly bytes: Buffer
-
-  constructor(
-    readonly name: string,
-    readonly ordinal: number,
-    readonly ordinalWidth: number,
-  ) {
-    this.bytes = Buffer.alloc(ordinalWidth)
-    for (let i = 0; i < ordinalWidth; i++) {
-      this.bytes[ordinalWidth - i - 1] = (ordinal >>> (i * 8)) & 0xff
-    }
-  }
-
-  toJSON(): string {
-    return this.name
-  }
-
-  toBytesSink(sink: BytesList): void {
-    sink.put(this.bytes)
-  }
-
-  toBytes(): Uint8Array {
-    return this.bytes
-  }
-}
-
-/*
- * @brief: Collection of Bytes objects, mapping bidirectionally
+import {
+  XrplDefinitionsBase,
+  FieldInstance,
+  Bytes,
+} from './xrpl-definitions-base'
+/**
+ * By default, coreTypes from the `types` folder is where known type definitions are initialized to avoid import cycles.
  */
-class BytesLookup {
-  constructor(types: Record<string, number>, readonly ordinalWidth: number) {
-    Object.entries(types).forEach(([k, v]) => {
-      this[k] = new Bytes(k, v, ordinalWidth)
-      this[v.toString()] = this[k]
-    })
-  }
+const DEFAULT_DEFINITIONS = new XrplDefinitionsBase(enums, {})
 
-  from(value: Bytes | string): Bytes {
-    return value instanceof Bytes ? value : (this[value] as Bytes)
-  }
-
-  fromParser(parser): Bytes {
-    return this.from(parser.readUIntN(this.ordinalWidth).toString())
-  }
-}
```

Comment:
`Bytes` and `ByteLookup` were added to `bytes.ts`: https://github.com/XRPLF/xrpl.js/blob/pass-through-encode/packages/ripple-binary-codec/src/enums/bytes.ts

* I added an error message when you try to add multiple bytes to the lookup with the same name because it results in invalid state. (It results in an ambiguous decoding since it could become either of the `named` values) - Tests failed when I did this by accident trying to figure out how it worked. - There's an error for both sides since the values are a 2-way lookup.